core/src/messages/agent.rs

```diff
@@ -241,6 +241,37 @@ pub struct SpawnRequest {
 
     /// Credentials used to fetch the image.
     pub credentials: Option<DockerCredentials>,
+
+    /// Resource limits
+    pub resource_limits: Option<ResourceLimits>,
```

Comment:
Rather than make this optional, is there a disadvantage to using

```
#[serde(default = "ResourceLimits::default")]
pub resource_limits: ResourceLimits,
```

This only makes sense if the defaults do not apply limits, which I think is the desired behavior.

Reply:
I didn't know I could do that :D